src/php/network/inet_pton.js

```diff
@@ -21,36 +21,44 @@ module.exports = function inet_pton (a) { // eslint-disable-line camelcase
     // Return if 4 bytes, otherwise false.
     return m.length === 4 ? m : false
   }
-  r = /^((?:[\da-f]{1,4}(?::|)){0,8})(::)?((?:[\da-f]{1,4}(?::|)){0,8})$/
 
   // IPv6
-  m = a.match(r)
-  if (m) {
-    // Translate each hexadecimal value.
-    for (j = 1; j < 4; j++) {
-      // Indice 2 is :: and if no length, continue.
-      if (j === 2 || m[j].length === 0) {
-        continue
+  if(a.length>39)return false;
+
+  m = a.split('::')
+
+  let res = ''
+  if(m.length>2)return false;	// :: can't be used more than once in IPv6.
+  for (j = 0; j < m.length; j++) {
+    if (m[j].length === 0) {	// Skip if empty.
+      continue
+    }
+    m[j] = m[j].split(':')
+    for (i = 0; i < m[j].length; i++) {
+
```

Comment:
here we could define `let hextet = hextets[i]` to not repeat `m[j][i]` all over the code.